syntax = "proto2";

import "protos/model.proto";
import "protos/hyperparams.proto";


message AdViSEBasicModel {
  extend Model {
    optional AdViSEBasicModel ext = 1433;
  }

  optional int32 hidden_units = 1 [default = 200];
  optional int32 output_units = 5 [default = 200];

  optional Hyperparams fc_hyperparams = 2;

  optional string glove_path = 3;

  optional string stmt_vocab_list_path = 4;

  optional float triplet_margin = 8 [default = 0.2];

  optional float dot_product_dropout_keep_prob = 9 [default = 0.5];

  optional int32 embedding_dims = 10 [default = 200];

  optional float max_norm = 12 [default = 6.0];

  optional bool add_image_as_a_roi = 11 [default = true];
}
